django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js

```diff
@@ -344,6 +359,85 @@
                 }
             });
         },
+        drawYearChoices: function(num) {
+            document.getElementById('nextPrevLinks' + num).style.display = 'none';
+            document.getElementById('calendarShortcuts' + num).style.display = 'none';
+            const cal_main = document.getElementById(DateTimeShortcuts.calendarDivName2 + num);
+            removeChildren(cal_main);	
+            const currentYear = DateTimeShortcuts.calendars[num].currentYear;
+            let yearClass = '';
+
+            // yearlist id used as identifier for ul element			    
+            const ul = quickElement('ul', cal_main, '', 'class', 'yearlist', 'id', 'yearlist' + num);
+            let yearLink = null;
+            for (let i = currentYear - 100; i < currentYear + 101; i++) {
+                if (i === currentYear) {
+                    yearClass = 'selected';
+                } else {
+                    yearClass = '';
+                }
+                const li = quickElement('li', ul, '');	
+                const link = quickElement('a', li, i, 'href', '#', 'class', yearClass);
+                if (yearClass === 'selected') {
+                    yearLink = link;
+                }
+                link.addEventListener('click', function(e) {
+                    e.preventDefault();
+                    // this loop remove className of ul links
+                    for (const childNode of ul.childNodes) {
+                        for (let k = 0; k < childNode.childNodes.length; k++) {
```

Comment:
You can enumerate by entries method like [here](https://stackoverflow.com/questions/34336960/what-is-the-es6-equivalent-of-python-enumerate-for-a-sequence).
I should have reviewed this one for first the review. I missed. I am sorry about that.

Reply:
That loop and a line was unnecessary so i deleted it, tnx for your suggests.